Servers/controllers/ssoAuth.ctrl.ts

```
@@ -0,0 +1,653 @@
+        }
+
+        // SECURITY: Safely extract and validate name information from Azure AD
+        let firstName = 'Unknown';
+        let lastName = 'User';
+
+        if (userInfo.name && typeof userInfo.name === 'string' && userInfo.name.trim().length > 0) {
+          const nameParts = userInfo.name.trim().split(' ');
+
+          // Sanitize first name
+          if (nameParts[0] && nameParts[0].length > 0) {
+            firstName = nameParts[0].replace(/[<>{}[\]\\\/\x00-\x1f\x7f]/g, '').substring(0, 50);
```

Biome (2.1.2)

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)